Library/Homebrew/gpg.rb

```diff
@@ -0,0 +1,47 @@
+require "utils"
+
+class Gpg
+  # Should ideally be using `GPGRequirement.new.gpg2`, etc to get path here but
+  # calling that directly leads to:
+  # requirement.rb:139:in `which_all': uninitialized constant Requirement::ORIGINAL_PATHS (NameError)
+  # when i.e. including the gpg syntax in wget. Not problematic if not used by formula code.
+  # For now, the path determination blob of code has been semi-modified for here.
+  # Look into this more.
+  def self.gpg
+    which("gpg") do |gpg|
+      gpg_short_version = Utils.popen_read(gpg, "--version")[/\d\.\d/, 0]
+      next unless gpg_short_version
+      Version.create(gpg_short_version.to_s) == Version.create("2.0")
+    end
+  end
+
+  def self.gpg2
+    which("gpg2") do |gpg2|
+      gpg2_short_version = Utils.popen_read(gpg2, "--version")[/\d\.\d/, 0]
+      next unless gpg2_short_version
+      Version.create(gpg2_short_version.to_s) == Version.create("2.0")
+    end
+  end
+
+  GPG_EXECUTABLE = gpg2 || gpg
+
+  def self.available?
+    File.exist?(GPG_EXECUTABLE.to_s) && File.executable?(GPG_EXECUTABLE)
```

Comment:
`which_all` (which I've switched back to locally) already does [some checks](https://github.com/Homebrew/brew/blob/604f4b6a0785e0295e548e6d033823303da19061/Library/Homebrew/utils.rb#L430) the file is executable & a file. This is mostly a smoke test so we actually have something for `available?`.